neutrino.go

```diff
@@ -1257,15 +1257,17 @@ func (s *ChainService) handleDonePeerMsg(state *peerState, sp *ServerPeer) {
 			s.connManager.Disconnect(sp.connReq.ID())
 		} else {
 			s.connManager.Remove(sp.connReq.ID())
-			go s.connManager.NewConnReq()
 		}
 		log.Debugf("Removed peer %s", sp)
-		return
+	} else {
+		// We'll always remove peers that are not persistent.
+		s.connManager.Remove(sp.connReq.ID())
 	}
 
-	// We'll always remove peers that are not persistent.
-	s.connManager.Remove(sp.connReq.ID())
-	go s.connManager.NewConnReq()
+	// Request a new connection if we still need more.
+	if state.Count() < MaxPeers {
+		go s.connManager.NewConnReq()
```

Comment:
this changes the logic slightly, by also requesting new connections after disconnecting a permanent peer. Which I guess is correct..? Why wasn't this done before?

Reply:
Much of the original code was lifted from `btcd`, which didn't care as much about aggressively maintaining the capped number of outbound peers. We should really start to consolidate some of this code though, as of recent anytime we make a change to this area of the codebase, then typically make a mirrored version for `btcd`.

Reply:
> Which I guess is correct..?

It's not correct, this is actually a regression. Fixed in the latest version. We don't request for permanent peers because those will be reconnected automatically by the connection manager.